x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -191,10 +192,26 @@ export function trainedModelsRoutes(
             mlLog.debug(e);
           }
 
-          const body = filterForEnabledFeatureModels(result, getEnabledFeatures());
+          try {
+            if (withOriginJobCheck) {
+              for (const model of result) {
+                if (typeof model.metadata?.analytics_config?.id === 'string') {
```

Comment:
We should probably be calling `getEnabledFeatures()` and not performing this check at all if `dfa` is `false`

Reply:
Updated in 7c06abd24a23cdfb98b34fba27bfdce6aee3e1ad